portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,13 +38,40 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<String> commentMessages = new ArrayList<String>();
-    commentMessages.add("This website looks real stylish!");
-    commentMessages.add("I also am a lover of chicken alfredo!");
-    commentMessages.add("Wow! How long did this take you?");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment").addSort("timeOfComment", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> listOfComments = new ArrayList<Comment>();
+    //Creates a Comment object for each entity that was previously ever posted. 
+    for (Entity commentEntity : results.asIterable()) {
+      Date timeOfComment = (Date) commentEntity.getProperty("timeOfComment");
+      String name = (String) commentEntity.getProperty("name");
+      String commentString = (String) commentEntity.getProperty("comment-string");
+      Comment comment = new Comment(timeOfComment, name, commentString);
+      listOfComments.add(comment);
+    }
+    //Changes the list of Comments into a JSON
     Gson gson = new Gson();
-    String json = gson.toJson(commentMessages);
+    String json = gson.toJson(listOfComments);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Date timeOfComment = new Date();
```

Comment:
Does new Date() automagically default to the current time?

Reply:
Looks like it does: 

https://docs.oracle.com/javase/8/docs/api/java/util/Date.html#Date--